### PR TITLE
Nav Redesign: Improve Plugins menu consistency

### PR DESCRIPTION
**Pull request by dsas**
Fixes https://github.com/Automattic/dotcom-forge/issues/6940 (see also pfsHM7-YW-p2)

## Proposed changes:
    
In order to improve the consistency menu of the Plugins menu for Simple and Atomic sites, this PR:
- Registers a "Plugins > Marketplace" menu for both Simple and Atomic that links to the `/plugins/:site`
- Register a "Add New Plugin" menu for Simple sites that links to `plugins.php?page=wpcom-install-plugin`. In the future, `plugin-install.php` might redirect here (see pMz3w-k1d-p2).

Screenshot 2024-05-29 at 13 18 59

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to your WP.com classic site
- Go to either Calypso or WP Admin for that site
- Check the Plugins menu
- Make sure there is a new "Plugins Marketplace" submenu that links to `/plugins/:site`
- For Simple sites, make sure there is a "Add New Plugin" submenu that links to `plugins.php?page=wpcom-install-plugin` and check the splash screen is displayed
- Make sure the rest of the submenus work as expected

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php [hunk 1]
```diff
@@ -599,6 +599,8 @@ function wpcom_add_scheduled_updates_menu() {
 
 	$domain = wp_parse_url( home_url(), PHP_URL_HOST );
 
+	// TODO: When looking at wordpress.com/plugins/:site on an atomic site this menu
```
Comment (dsas) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: This is the only thing that needs fixing I think 🤔

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php [hunk 2]
```diff
@@ -611,26 +613,75 @@ function wpcom_add_scheduled_updates_menu() {
...
+			$found_plugin_menu = true;
+		}
+
+		if ( ! $found_plugin_install_submenu ) {
+			add_submenu_page(
+				'plugins.php',
+				__( 'Add New Plugin', 'jetpack-mu-wpcom' ),
+				__( 'Add New Plugin', 'jetpack-mu-wpcom' ),
+				'manage_options', // Roughly means "is a site admin"
+				// Don't have a plugin-install.php page on Simple sites yet
```
Comment (dsas) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: Opened D149817-code to attempt that. Maybe though we can just change this link to use /wp-admin/plugins.php on simple sites?

Comment (mmtr) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: I think that could cause the menu highlight issues that triggered these changes 😅 

One alternative would be to not include an "Add New Plugin" menu for now, until we figure out a way forward for Simple sites

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php [hunk 3]
```diff
@@ -611,26 +613,75 @@ function wpcom_add_scheduled_updates_menu() {
...
 	}
+	global $menu;
+
+	$is_simple_site = defined( 'IS_WPCOM' ) && IS_WPCOM;
+
+	$found_plugin_menu            = false;
+	$found_plugin_install_submenu = false;
+	foreach ( $menu as &$menu_item ) {
+		if ( 'plugins.php' === $menu_item[2] ) {
+			$found_plugin_menu = true;
```
Comment (mmtr) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: I'm failing to see under which circumstances this menu can exist for Simple sites, so I'm considering to simplify the logic here (and have something similar to what we had before).

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php (hunk 1 above)
Comment (mmtr) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: Handled in Calypso: https://github.com/Automattic/wp-calypso/pull/91200

**Comment by mmtr** [comment]
> Tests / Static analysis (pull_request) Failing after 4m

Updated the stubs in D150355-code but they are not updated in Jetpack. Asked in p1716983860931499-slack-CDLH4C1UZ

EDIT: Solved

**Review by taipeicoder — APPROVED** [review]
Works as intended 👍 

| Simple Classic | Atomic Classic |
| --- | --- | 
| Screenshot 2024-05-30 at 11 11 29 AM | Screenshot 2024-05-30 at 11 10 47 AM|

Simple Classic:
- Install Plugins links to /wp-admin/plugins.php (SA-only)
- Add New Plugin links to /wp-admin/plugins.php?page=wpcom-install-plugin
- Marketplace links to https://wordpress.com/plugins/:site

Atomic Classic
- Install Plugins links to /wp-admin/plugins.php 
- Add New Plugin links to /wp-admin/plugin-install.php
- Marketplace links to https://wordpress.com/plugins/:site
- Scheduled Updates links to https://wordpress.com/plugins/scheduled-updates/:site

**Pull request merged**